sdk/python/feast/infra/offline_stores/contrib/spark_offline_store/spark_source.py

```diff
@@ -180,11 +206,70 @@ def get_table_query_string(self) -> str:
             logger.exception(
                 "Spark read of file source failed.\n" + traceback.format_exc()
             )
-        tmp_table_name = get_temp_entity_table_name()
+        tmp_table_name = "feast_entity_df_" + uuid.uuid4().hex
```

Comment:
We can use same method

Reply:
The method is inside a utility file that imports FeatureView, creating a circular dependency. SparkSource is a DataSource, which should not ever depend on FeatureView, so this was a bug in SparkSource. I thought about moving the method out to a second new utility file, but decided it would be cheaper and less intrusive to the feast code to just use the one liner.